src/core/server/elasticsearch/elasticsearch_config.ts

```diff
@@ -195,5 +212,14 @@ export class ElasticsearchConfig {
       ...rawConfig.ssl,
       certificateAuthorities,
     };
+
+    if (this.username === 'elastic' && log !== undefined) {
+      // logger is optional / not used during tests
+      // TODO: logger can be removed when ISSUE#40255 is resolved to support deprecations in NP config service
+      log.warn(
+        `Setting the elasticsearch username to "elastic" in production mode is deprecated. You should use the "kibana" user instead.`,
```

Comment:
nit: most users of Kibana likely won't know what "production mode" is, perhaps we should remove this from the deprecation warning text?

Reply:
Yep